### PR TITLE
Feat/4851 highlight mentions on browser

### PR DESCRIPTION
**Pull request by MariePrandstetter**
#4851

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend/src/container/Tracim.jsx [hunk 1]
```diff
@@ -363,28 +373,26 @@ export class Tracim extends React.Component {
...
       document.title = newHeadTitle
     }
 
-    if (
-      !hasNotificationNotReadCountChanged ||
-      (prevNotificationNotReadCount > 1 && notificationNotReadCount > 1)
-    ) return
-
-    toggleFavicon(notificationNotReadCount > 0)
+    if ((hasUnreadMentionCountChanged && (prevUnreadMentionCount * unreadMentionCount === 0)) || (hasUnreadNotificationCountChanged && (prevUnreadNotificationCount * unreadNotificationCount) === 0)) {
```
Comment (gbonaspetti) [inline frontend/src/container/Tracim.jsx:393]: Shouldn't it be `+` instead of `*`?

File: frontend/src/reducer/notificationPage.js [hunk 2]
```diff
@@ -65,38 +67,46 @@ export default function notificationPage (state = defaultNotificationsObject, ac
...
-        notificationNotReadCount: state.notificationNotReadCount + 1
+        unreadMentionCount: newUnreadMentionCount,
+        unreadNotificationCount: state.unreadNotificationCount + 1
       }
     }
 
     case `${READ}/${NOTIFICATION}`: {
       const notification = state.list.find(notification => notification.id === action.notificationId && !notification.read)
       if (!notification) return state
+      const newUnreadMentionCount = notification.type === 'mention.created' ? state.unreadMentionCount - 1 : state.unreadMentionCount
```
Comment (gbonaspetti) [inline frontend/src/reducer/notificationPage.js:86]: Should we have a const for `'mention.created'`?

File: frontend/src/util/helper.js [hunk 3]
```diff
@@ -212,7 +212,7 @@ export const toggleFavicon = (hasNewNotification) => {
       )
       // FIXME - GM - 2020/08/18 - Replace this hardcoded values to webpack variables
       // https://github.com/tracim/tracim/issues/2098
-      context.fillStyle = '#3F9FF7'
+      context.fillStyle = hasUnreadMention ? '#dc3545' : '#3F9FF7'
```
Comment (gbonaspetti) [inline frontend/src/util/helper.js:215]: I think you can use `#ff8a00` to be coherent with #4858 

```suggestion
      context.fillStyle = hasUnreadMention ? '#ff8a00' : '#3F9FF7'
```

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend/src/container/Tracim.jsx (hunk 1 above)
Comment (MariePrandstetter) [reply frontend/src/container/Tracim.jsx:393]: yes that makes sense

File: frontend/src/reducer/notificationPage.js (hunk 2 above)
Comment (MariePrandstetter) [reply frontend/src/reducer/notificationPage.js:86]: I created a mentionCreated const

**Review by gbonaspetti — APPROVED** [review]

**Comment by charline-rageade** [comment]
Tested on branch

**Pull request merged**